src/tools/miri/tests/fail/memleak.stderr

```diff
@@ -1,6 +1,7 @@
 The following memory was leaked: ALLOC (Rust heap, size: 4, align: 4) {
 $stripped$
-}
+}{"message":"allocALLOC was allocated RUSTLIB/alloc/src/alloc.rs","byte_start":3023,"byte_end":3066,"line_start":95,"line_end":95,"column_start":14,"column_end":57,"is_primary":true,"text":[{"text":"    unsafe { __rust_alloc(layout.size(), layout.align()) }","highlight_start":14,"highlight_end":57}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"inside `std::alloc::alloc` at RUSTLIB/alloc/src/alloc.rs:LL:CC","code":null,"level":"note","spans":[],"children":[],"rendered":null},{"message":"inside `std::alloc::Global::alloc_impl` at RUSTLIB/alloc/src/alloc.rs:LL:CC","code":null,"level":"note","spans":[],"children":[],"rendered":null},{"message":"inside `<std::alloc::Global as std::alloc::Allocator>::allocate` at RUSTLIB/alloc/src/alloc.rs:LL:CC","code":null,"level":"note","spans":[],"children":[],"rendered":null},{"message":"inside `alloc::alloc::exchange_malloc` at RUSTLIB/alloc/src/alloc.rs:LL:CC","code":null,"level":"note","spans":[],"children":[],"rendered":null},{"message":"inside `std::boxed::Box::<i32>::new` at RUSTLIB/alloc/src/boxed.rs:LL:CC","code":null,"level":"note","spans":[],"children":[],"rendered":null},{"message":"inside `main`","code":null,"level":"note","spans":[{"file_name":"$DIR/memleak.rs","byte_start":142,"byte_end":154,"line_start":5,"line_end":5,"column_start":22,"column_end":34,"is_primary":true,"text":[{"text":"    std::mem::forget(Box::new(42));","highlight_start":22,"highlight_end":34}],"label":null,"suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[],"rendered":null}],"rendered":"note: allocALLOC was allocated here:/n  --> RUSTLIB/alloc/src/alloc.rs:LL:CC/n   |/nLL |     unsafe { __rust_alloc(layout.size(), layout.align()) }/n   |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^/n   |/n   = note: inside `std::alloc::alloc` at RUSTLIB/alloc/src/alloc.rs:LL:CC/n   = note: inside `std::alloc::Global::alloc_impl` at RUSTLIB/alloc/src/alloc.rs:LL:CC/n   = note: inside `<std::alloc::Global as std::alloc::Allocator>::allocate` at RUSTLIB/alloc/src/alloc.rs:LL:CC/n   = note: inside `alloc::alloc::exchange_malloc` at RUSTLIB/alloc/src/alloc.rs:LL:CC/n   = note: inside `std::boxed::Box::<i32>::new` at RUSTLIB/alloc/src/boxed.rs:LL:CC/nnote: inside `main`/n  --> $DIR/memleak.rs:LL:CC/n   |/nLL |     std::mem::forget(Box::new(42));/n   |                      ^^^^^^^^^^^^/n/n"}
```

Comment:
looks like there's a newline missing between the `}{` at the start and thus this doesn't get rendered but just passed through

Reply:
I'm not sure this approach is compatible with the existing leak report formatting: https://github.com/rust-lang/rust/pull/109061#discussion_r1142826178

In particular, if the `Formatter` we are writing to inside the leak reports decides to get fancy with buffering, we're hosed.